src/main/java/io/icker/factions/ui/AdminGui.java

```diff
@@ -0,0 +1,252 @@
+                        user.setSpoof(null);
+                        new Message("Cleared spoof").send(player, false);
+                        return;
+                    }
+                    this.execSpoof();
+                })
+        );
+
+        // Audit icon
+        this.setSlot(indexes.get(3), new GuiElementBuilder(Items.PLAYER_HEAD)
+                .setSkullOwner(Icons.GUI_BOOK)
+                .setName(Text.literal("Run an audit"))
+                .setLore(List.of(Text.literal("Click to check saved data to ensure there's no issues.").setStyle(Style.EMPTY.withItalic(false).withColor(Formatting.GRAY))))
+                .setCallback((index, clickType, actionType) -> {
+                    GuiInteract.playClickSound(player);
+                    for (int i = 0; i < 4; i++) {
+                        Claim.audit();
+                        Faction.audit();
+                        User.audit();
+                    }
+
+                    new Message("Successful audit").send(player, false);
+                })
+        );
+
+        if (indexes.size() > 4) {
+            // Dynmap reload icon
+            this.setSlot(indexes.get(4), new GuiElementBuilder(Items.PLAYER_HEAD)
+                    .setSkullOwner(Icons.GUI_EARTH_RELOAD)
+                    .setName(Text.literal("Reload DynMap markers"))
+                    .setLore(List.of(Text.literal("Reloads dynmap markers.").setStyle(Style.EMPTY.withItalic(false).withColor(Formatting.GRAY))))
+                    .setCallback((index, clickType, actionType) -> {
+                        GuiInteract.playClickSound(player);
+                        FactionsMod.dynmap.reloadAll();
+                        new Message("Reloaded dynmap marker").send(player, false);
+                    })
+            );
+        }
+
+        this.open();
+    }
+
+    private void execSpoof() {
+        InputGui inputGui = new InputGui(player);
+
+        inputGui.setTitle(Text.literal("Specify a player..."));
+        inputGui.setDefaultInputValue("Player Name");
+
+        inputGui.returnBtn.setCallback(defaultReturn);
+        inputGui.confirmBtn.setCallback(
+                (index, clickType, actionType) -> {
+                    GuiInteract.playClickSound(player);
+
+                    String input = inputGui.getInput();
+
+                    if (!input.matches("^[a-zA-Z0-9_]{2,16}$")) {
+                        inputGui.showErrorMessage(Text.literal(String.format("Invalid name %s!", input)).formatted(Formatting.RED), index);
+                        return;
+                    }
+                    User target;
+                    Optional<GameProfile> profile;
+                    if ((profile = player.getServer().getUserCache().findByName(input)).isPresent()) {
+                        target = User.get(profile.get().getId());
+                    } else {
+                        try {
+                            target = User.get(UUID.fromString(input));
```

Comment:
Any valid UUIDs are filtered by the regex above, so this will never do anything. I don't think the regex is even needed because it's already handled below.